src/odemis/driver/hamamatsurx.py

```diff
@@ -932,29 +930,36 @@ def __init__(self, name, role, port, host, children=None, dependencies=None, dae
         # TODO want to check if we want to start app invisible (sVisible = False)
 
         self.AppStart()  # start HPDTA software  # Note: comment out for testing in order to not start a new App
+        try:
+            children = children or {}
+            dependencies = dependencies or {}
 
-        children = children or {}
-        dependencies = dependencies or {}
+            try:
+                ckwargs = children["readoutcam"]
+            except Exception:
+                raise ValueError("Required child readoutcam not provided")
 
-        try:
-            kwargs = children["readoutcam"]
-        except Exception:
-            raise ValueError("Required child readoutcam not provided")
-        self._readoutcam = ReadoutCamera(parent=self, spectrograph=dependencies.get("spectrograph"),
-                                         daemon=daemon, **kwargs)
-        self.children.value.add(self._readoutcam)  # add readoutcam to children-VA
-        try:
-            kwargs = children["streakunit"]
-        except Exception:
-            raise ValueError("Required child streakunit not provided")
-        self._streakunit = StreakUnit(parent=self, daemon=daemon, **kwargs)
-        self.children.value.add(self._streakunit)  # add streakunit to children-VA
-        try:
-            kwargs = children["delaybox"]
+            self._readoutcam = ReadoutCamera(parent=self, spectrograph=dependencies.get("spectrograph"),
+                                             daemon=daemon, **ckwargs)
+            self.children.value.add(self._readoutcam)  # add readoutcam to children-VA
+            try:
+                ckwargs = children["streakunit"]
+            except Exception:
+                raise ValueError("Required child streakunit not provided")
+            self._streakunit = StreakUnit(parent=self, daemon=daemon, **ckwargs)
+            self.children.value.add(self._streakunit)  # add streakunit to children-VA
+            try:
+                ckwargs = children["delaybox"]
+            except Exception:
+                raise ValueError("Required child delaybox not provided")
+            self._delaybox = DelayGenerator(parent=self, daemon=daemon, **ckwargs)
+            self.children.value.add(self._delaybox)  # add delaybox to children-VA
         except Exception:
-            raise ValueError("Required child delaybox not provided")
-        self._delaybox = DelayGenerator(parent=self, daemon=daemon, **kwargs)
-        self.children.value.add(self._delaybox)  # add delaybox to children-VA
+            try:
```

Comment:
Add a comment why you try to end the app here. To prevent having started the App, failing to connect for some reason and then having the app still running when trying again to start odemis.